### PR TITLE
cropcalStreamMod compiler error bug fix

### LINKED ISSUE #2232: COMPILER ERROR problem in cropcalStreamMod
### Brief summary of bug

I am having an annoying compiler error running on my personal computer (Mac, with compiling with GNU 13.2.0).

The compiler error seems to originate somewhere in the cropcalStreamMod:

```
f951: internal compiler error: Segmentation fault: 11
Please submit a full bug report, with preprocessed source (by using -freport-bug).
See <https://github.com/Homebrew/homebrew-core/issues> for instructions.
gmake: *** [/Users/afoster/Documents/ncar/NEON/NEON_runs/cases/BONA_run_test/Tools/Makefile:975: cropcalStreamMod.o] Error 1
gmake: *** Waiting for unfinished jobs....
```
### General bug information

I tested `ctsm5.1.dev146` and it compiled, whereas 147 and above do not. 

At @billsacks's suggestion I am going to try changing the `use ESMF` statement to have `only`s for what the module specifically needs, as this *could* be the problem. Otherwise I will need to start commenting out blocks of the code to figure out what the offending lines could be.

**CTSM version you are using:** `ctsm5.1.dev147` and above

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** Using latest (or above 12.2) gnu fortran to compile

### PR DESCRIPTION
**Pull request by adrifoster**
### Description of changes

Fixes a compiler error (for GNU 13.2) within cropcalStreamMod

### Specific notes

Simple fix was to change whole-array assignments/references for the `starts` and `ends` arrays to specifically reference bounds (`begp` and `endp`).

Contributors other than yourself, if any: @billsacks 

CTSM Issues Fixed (include github issue #): Closes #2232

Are answers expected to change (and if so in what way)? No

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any:

Successfully compiles with GNU 13.2

### CODE REVIEW

File: src/cpl/share_esmf/cropcalStreamMod.F90 [hunk 1]
```diff
@@ -248,12 +249,12 @@ subroutine cropcal_advance( bounds )
...
     ! Advance crop calendar streams
     !
-    ! !USES:
+    !USES:
     use clm_time_manager , only : get_curr_date
     use dshr_strdata_mod , only : shr_strdata_advance
     !
     ! !ARGUMENTS:
-    type(bounds_type), intent(in)    :: bounds
+     type(bounds_type), intent(in)    :: bounds
```
Comment (samsrabin) [inline src/cpl/share_esmf/cropcalStreamMod.F90:257]: Extraneous leading whitespace.

File: src/cpl/share_esmf/cropcalStreamMod.F90 [hunk 2]
```diff
@@ -298,22 +299,22 @@ end subroutine cropcal_advance
...
+     use clm_time_manager, only : get_curr_days_per_year
+     use pftconMod       , only : pftname
+     use dshr_methods_mod , only : dshr_fldbun_getfldptr
+     !
+     ! !ARGUMENTS:
+     implicit none
+     type(bounds_type)      , intent(in)    :: bounds
+     integer                , intent(in)    :: num_pcropp        ! number of prog. crop patches in filter
+     integer                , intent(in)    :: filter_pcropp(:)  ! filter for prognostic crop patches
+     type(crop_type)        , intent(inout) :: crop_inst
```
Comment (samsrabin) [inline src/cpl/share_esmf/cropcalStreamMod.F90:317]: Extraneous leading whitespace at lines 302-317.

File: src/cpl/share_esmf/cropcalStreamMod.F90 [hunk 3]
```diff
@@ -397,17 +402,17 @@ subroutine cropcal_interp(bounds, num_pcropp, filter_pcropp, crop_inst)
...
+           if (any(ends(begp:endp,2:mxsowings) <= ends(begp:endp,1:mxsowings-1) .and. &
+                   ends(begp:endp,2:mxsowings) >= 1)) then
                write(iulog, *) 'Sowing window inputs must be ordered such that end dates are monotonically increasing.'
                call ESMF_Finalize(endflag=ESMF_END_ABORT)
            end if
        end if
 
        ! Handle invalid sowing window values
-       if (any(starts < 1 .or. ends < 1)) then
+        if (any(starts(begp:endp,:) < 1 .or. ends(begp:endp,:) < 1)) then
```
Comment (samsrabin) [inline src/cpl/share_esmf/cropcalStreamMod.F90:413]: Extraneous leading whitespace.

File: src/cpl/share_esmf/cropcalStreamMod.F90 [hunk 4]
```diff
@@ -422,11 +427,11 @@ subroutine cropcal_interp(bounds, num_pcropp, filter_pcropp, crop_inst)
                call ESMF_Finalize(endflag=ESMF_END_ABORT)
 
            ! Fail if a sowing window start date is prescribed without an end date (or vice versa)
-           else if (any((starts >= 1 .and. ends < 1) .or. (starts < 1 .and. ends >= 1))) then
+           else if (any((starts(begp:endp,:) >= 1 .and. ends(begp:endp,:) < 1) .or. (starts(begp:endp,:) < 1 .and. ends(begp:endp,:) >= 1))) then
                write(iulog, *) 'Every prescribed sowing window start date must have a corresponding end date.'
                call ESMF_Finalize(endflag=ESMF_END_ABORT)
            end if
-       end if
+        end if
```
Comment (samsrabin) [inline src/cpl/share_esmf/cropcalStreamMod.F90:434]: Extraneous leading whitespace.

File: src/cpl/share_esmf/cropcalStreamMod.F90 [hunk 5]
```diff
@@ -54,15 +55,15 @@ subroutine cropcal_init(bounds)
...
+    !USES:
     use shr_mpi_mod      , only : shr_mpi_bcast
     use clm_nlUtilsMod   , only : find_nlgroup_name
     use lnd_comp_shr     , only : mesh, model_clock
     use dshr_strdata_mod , only : shr_strdata_init_from_inline
     use controlMod       , only : NLFilename
     !
     ! !ARGUMENTS:
-    type(bounds_type), intent(in) :: bounds          ! bounds
+     type(bounds_type), intent(in) :: bounds          ! bounds
```
Comment (samsrabin) [inline src/cpl/share_esmf/cropcalStreamMod.F90:66]: Extraneous leading whitespace.

**Review by samsrabin — APPROVED** [review]
Looks good, thanks! I just found some lines where some extra whitespace seems to have been inserted, leading to apparent diffs that really aren't. If you could revert that, that'd be great; no need for me to re-review so I'll approve now.

Thanks again!

**Comment by adrifoster** [comment]
> Looks good, thanks! I just found some lines where some extra whitespace seems to have been inserted, leading to apparent diffs that really aren't. If you could revert that, that'd be great; no need for me to re-review so I'll approve now.

sorry about that, fixed now!

**Pull request merged**